File: 2025-06-04/Minimum_Size_Subarray_Sum.java
package patterns.sliding_window;

public class Minimum_Size_Subarray_Sum {
    public static int minSubArrayLen(int target, int[] nums) {
        int left = 0, sum = 0, result = Integer.MAX_VALUE;
        for(int right=0;right<nums.length;right++) {
            sum += nums[right];
            while(sum >= target) {
                result = Math.min(result, right - left + 1);
                sum -= nums[left++];
            }
        }

        return result == Integer.MAX_VALUE ? 0 : result;
    }
    public static void main(String[] args) {
        int[] arr = {1,2,3,4,5};
        System.out.println(minSubArrayLen(15, arr));
    }
}

File: 2026-01-31/Minimum_Size_Subarray_Sum.py
from typing import List


class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        total = sum(nums)
        n = len(nums)

        if total < target:
            return 0

        if total == target:
            return len(nums)

        rSum = l = 0
        minLen = n
        for r in range(n + 1):
            while rSum >= target:
                minLen = min(minLen, r - l)
                rSum -= nums[l]
                l += 1

            if r < n:
                rSum += nums[r]

        return minLen


sol = Solution()
print(sol.minSubArrayLen(15, [5,1,3,5,10,7,4,9,2,8]))